docs/principles.md

```diff
@@ -0,0 +1,69 @@
+# OmnICU Design Principles & Decisions
+
+## Guiding principles
+
+### Immutable objects
+
+Discussion about builders, constructors. But no object is to be mutated after creation.
+
+### No I/O in the core library
+
+Leave I/O to the wrapper or clients. Pass data as reference to the methods.
+Clients can mmap, async fetch, package data with the library...
+
+### No internal threading
+
+Both Clojure and Wasm support multithreading but we don’t have a need for it in the i18n realm.
```

Comment:
Should we change "Wasm" to "Rust" given the adjustment in focus since the doc was drafted?

Reply:
Doc looks good otherwise.

Reply:
I assume you meant Closure to Rust, instead of Wasm to Rust.

Reply:
Hmm, changing "Clojure" to "Rust" does make more sense.  I was thinking about Wasm, but that should be included alongside Rust.  I guess it's all a moot point since we're describing a feature that we don't need to enter into the core algorithmic code.